sdk/communication/azure-communication-chat/azure/communication/chat/_shared/user_credential.py

```diff
@@ -3,84 +3,121 @@
 # Licensed under the MIT License. See License.txt in the project root for
 # license information.
 # --------------------------------------------------------------------------
-from threading import Lock, Condition
-from datetime import timedelta
-from typing import ( # pylint: disable=unused-import
-    cast,
-    Tuple,
-)
 
+from threading import Lock, Condition, Timer
+from datetime import timedelta
+from typing import Any
+import six
 from .utils import get_current_utc_as_int
-from .user_token_refresh_options import CommunicationTokenRefreshOptions
+from .utils import create_access_token
 
 
 class CommunicationTokenCredential(object):
     """Credential type used for authenticating to an Azure Communication service.
-    :param str token: The token used to authenticate to an Azure Communication service
-    :keyword token_refresher: The token refresher to provide capacity to fetch fresh token
+    :param str token: The token used to authenticate to an Azure Communication service.
+    :keyword callable token_refresher: The async token
+     refresher to provide capacity to fetch a fresh token. The returned token must be valid (expiration
+     date must be in the future).
+    :keyword bool refresh_proactively: Whether to refresh the token proactively or not.
     :raises: TypeError
     """
 
     _ON_DEMAND_REFRESHING_INTERVAL_MINUTES = 2
-
-    def __init__(self,
-                 token,  # type: str
-                 **kwargs
-                 ):
-        token_refresher = kwargs.pop('token_refresher', None)
-        communication_token_refresh_options = CommunicationTokenRefreshOptions(token=token,
-                                                                               token_refresher=token_refresher)
-        self._token = communication_token_refresh_options.get_token()
-        self._token_refresher = communication_token_refresh_options.get_token_refresher()
+    _DEFAULT_AUTOREFRESH_INTERVAL_MINUTES = 10
```

Comment:
I thought per the previous archboard review, that the proactive refresh would be done according to a percentage of the remaining validity rather than an absolute time interval?

Reply:
Ah - it looks like we are, and this is more like a default window

Reply:
yeah, it is the default expiring soon window. Here is the design of how the autorefresh mechanism works https://github.com/Azure/azure-sdk/issues/3712#issuecomment-1011186174